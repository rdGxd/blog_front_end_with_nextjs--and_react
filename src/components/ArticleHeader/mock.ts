import { ArticleHeaderProps } from ".";
import { data } from "../../api/dados.json";

const { title, excerpt, cover, author, categories, createdAt } = data.posts.data[0].attributes;

const mockArticleHeader = {
  title,
  author,
  categories,
  cover,
  createdAt,
  excerpt,
} as ArticleHeaderProps;

export default mockArticleHeader;
