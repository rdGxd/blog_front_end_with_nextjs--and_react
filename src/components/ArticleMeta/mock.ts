import { ArticleMetaProps } from ".";
import { data } from "../../api/dados.json";

const { categories, author, createdAt } = data.posts.data[0].attributes;

const mockArticleMeta = {
  createdAt,
  categories: categories.data,
  author,
} as ArticleMetaProps;

export default mockArticleMeta;
