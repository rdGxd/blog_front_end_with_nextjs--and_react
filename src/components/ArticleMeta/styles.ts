import styled, { css } from "styled-components";

export const Wrapper = styled.div`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.small};
    color: ${theme.colors.darkerGray};
    font-style: italic;

    .categories span::after {
      content: ", ";
    }

    .categories span:last-child::after {
      content: "";
    }

    a:hover {
      filter: brightness(50%);
    }
  `}
`;
