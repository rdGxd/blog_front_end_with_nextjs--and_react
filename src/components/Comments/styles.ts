import styled, { css } from "styled-components";

export const Wrapper = styled.div`
  ${({ theme }) => css`
    max-width: ${theme.sizes.content};
    width: 100%;
    margin: ${theme.spacings.large} 0 auto;
    padding: 0 ${theme.spacings.large};
  `}
`;
