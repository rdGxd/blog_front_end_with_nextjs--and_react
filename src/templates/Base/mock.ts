import { BaseProps } from ".";
import { data } from "../../api/dados.json";

export default {
  settings: data.setting.data,
  children: "Ol√° mundo",
} as unknown as BaseProps;
