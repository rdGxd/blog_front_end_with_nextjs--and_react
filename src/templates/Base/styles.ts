import styled, { css } from "styled-components";

export const Wrapper = styled.div``;

export const HeaderContainer = styled.header`
  ${({ theme }) => css`
    ${theme.spacings.xhuge}
  `}
`;

export const ContentContainer = styled.div`
  ${({ theme }) => css`
    max-width: 120rem;
    width: 100%;
    margin: 0 auto;
    margin-bottom: ${theme.spacings.xhuge};
  `}
`;

export const FooterContainer = styled.footer`
  ${({ theme }) => css`
    max-width: 120rem;
    width: 100%;
    margin: 0 auto;
    padding: ${theme.spacings.large};
  `}
`;
