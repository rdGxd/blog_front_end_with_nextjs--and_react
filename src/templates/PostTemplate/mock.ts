import { PostTemplateProps } from ".";
import { data } from "../../api/dados.json";

export default {
  settings: data.setting,
  post: data.posts.data[0].attributes,
} as unknown as PostTemplateProps;
