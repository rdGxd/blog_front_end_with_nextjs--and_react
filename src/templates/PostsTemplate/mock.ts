import { PostsTemplateProps } from ".";
import { data } from "../../api/dados.json";

export default {
  settings: data.setting.data,
  posts: data.posts.data,
} as PostsTemplateProps;
