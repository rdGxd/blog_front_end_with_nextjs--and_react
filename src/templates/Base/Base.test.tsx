import { screen } from "@testing-library/react";
import { Base } from ".";
import { renderTheme } from "../../styles/render-theme";
import mock from "./mock";

describe("<Base />", () => {
  it("should render base elements", () => {
    const { debug } = renderTheme(<Base {...mock} />);

    expect(screen.getByRole("img", { name: "Rodrigo Silva - Blog do Rodrigo" })).toBeInTheDocument();
    debug();
    expect(screen.getByLabelText("Open or close menu")).toBeInTheDocument();

    expect(screen.getByText("texto do footer")).toBeInTheDocument();

    expect(screen.getByLabelText("Go to top")).toBeInTheDocument();
  });

  it("should render base elements", () => {
    const { container } = renderTheme(<Base {...mock} />);

    expect(container).toMatchSnapshot();
  });
});
