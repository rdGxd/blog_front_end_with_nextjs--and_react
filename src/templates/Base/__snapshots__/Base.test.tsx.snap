// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Base /> > should render base elements 1`] = `
.c1 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c17 {
  position: fixed;
  background: #000000;
  color: #FFFFFF;
  display: flex;
  align-items: center;
  justify-content: center;
  width: 4rem;
  height: 4rem;
  bottom: 2rem;
  right: 2rem;
  z-index: 6;
  opacity: 0.7;
}

.c16 {
  font-size: 2.4rem;
}

.c16 p {
  margin: 2.4rem 0;
}

.c16 a:visited,
.c16 a:link {
  color: #dc143c;
  text-decoration: none;
  transition: all 300ms ease-in-out;
}

.c16 a:hover {
  filter: brightness(80%);
}

.c16 code {
  font-family: monospace;
  color: #dc143c;
  font-size: 1.6rem;
  background: #DDDDDD;
  padding: 0.2rem;
  margin: 0.2rem;
}

.c16 pre {
  background: #000000;
  padding: 2.4rem;
  font-family: monospace;
  color: #FFFFFF;
  margin: 2.4rem 0;
  width: 100%;
  overflow-x: auto;
  font-size: 1.6rem;
}

.c16 pre code {
  color: inherit;
  background: inherit;
}

.c16 img {
  max-width: 100%;
}

.c16 .image {
  background: #DDDDDD;
  line-height: 0;
  margin: 2.4rem 0;
}

.c16 .image figcaption {
  text-align: center;
  font-size: 1.6rem;
  padding: 1.6rem;
  line-height: 1.3;
}

.c16 .image-style-side {
  max-width: 50%;
  float: right;
  margin: 2.4rem 0;
}

.c16 hr {
  border: none;
  border-bottom: 0.1rem solid #DDDDDD;
}

.c16 ul,
.c16 ol {
  margin: 2.4rem 4.0rem;
}

.c16 .table {
  width: 100%;
  border-collapse: collapse;
  margin: 2.4rem 0;
}

.c16 table {
  width: 100%;
  border-collapse: collapse;
}

.c16 table td,
.c16 table th {
  padding: 1.6rem;
  border: 0.1rem solid #DDDDDD;
}

.c16 blockquote {
  border-left: 0.5rem solid #dc143c;
  color: #AAAAAA;
  filter: brightness(80%);
  padding-left: 2.4rem;
  font-style: italic;
  margin: 2.4rem;
}

.c14 {
  text-align: center;
  border-top: 0.1rem solid #DDDDDD;
}

.c14 a {
  color: inherit;
  text-decoration: none;
}

.c14 .c15 {
  font-size: 1.6rem;
}

.c6 {
  color: #000000;
  line-height: 1.2;
  font-size: 2.4rem;
  text-transform: uppercase;
}

.c11 {
  color: #000000;
  line-height: 1.2;
  font-size: 2.4rem;
  text-transform: none;
}

.c7 {
  display: flex;
  align-items: center;
  text-decoration: none;
  color: inherit;
}

.c7 img {
  width: 15rem;
  height: 15rem;
  border-radius: 50%;
}

.c9 {
  display: flex;
  align-items: center;
  justify-content: center;
  padding-bottom: 6.4rem;
  width: 100%;
  max-width: 96rem;
  color: #333333;
  font-size: 1.6rem;
  margin: 0 auto;
}

.c9 .c5 {
  margin: 0 0 calc(1.6rem -1rem);
}

.c10 {
  display: flex;
  flex-flow: column wrap;
  margin-left: 3.2rem;
  max-width: 48rem;
}

.c8 {
  display: block;
  color: #FFFFFF;
  text-decoration: none;
  margin-bottom: 1.6rem;
  font-size: 1.8rem;
  border-right: 0.5rem solid #000000;
  transition: all 300ms ease-in-out;
}

.c8:hover {
  border-right: 0.5rem solid #dc143c;
  color: #dc143c;
}

.c2 {
  display: flex;
  background: #000000;
  padding: 3.2rem;
  position: fixed;
  z-index: 1;
  width: 100%;
  max-width: 32rem;
  height: 100vh;
  top: 0;
  left: 0;
  transition: all 300ms ease-in-out;
  overflow-y: auto;
  left: -35rem;
  overflow-y: hidden;
}

.c3 {
  margin: auto;
  width: 100%;
}

.c4 .c5 {
  display: flex;
  justify-content: center;
  margin: 0;
  margin-bottom: 4.0rem;
}

.c4 .c5 img {
  border: 0.5rem solid #dc143c;
}

.c0 {
  position: fixed;
  top: 2.4rem;
  color: #FFFFFF;
  background: #000000;
  z-index: 2;
  width: 3rem;
  height: 3rem;
  left: 26rem;
  left: 1rem;
  color: #FFFFFF;
}

.c12 {
  max-width: 120rem;
  width: 100%;
  margin: 0 auto;
  margin-bottom: 6.4rem;
}

.c13 {
  max-width: 120rem;
  width: 100%;
  margin: 0 auto;
  padding: 3.2rem;
}

@media (max-width: 768px) {
  .c16 {
    font-size: 2rem;
  }

  .c16 .image-style-side {
    max-width: 100%;
    float: none;
    margin: 0;
  }
}

@media (max-width: 768px) {
  .c2 {
    left: -32rem;
  }
}

@media (max-width: 768px) {
  .c0 {
    left: -0.1rem;
  }
}

<div>
  <div
    class=""
  >
    <a
      aria-label="Open or close menu"
      class="c0"
      href="#"
      title="Open or close menu"
    >
      <svg
        aria-hidden="true"
        aria-label="Open menu"
        class="c1"
        fill="currentColor"
        focusable="false"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
        />
      </svg>
    </a>
    <div
      aria-hidden="true"
      class="c2"
    >
      <nav
        class="c3"
      >
        <div
          class="c4"
        >
          <h1
            class="c5 c6"
          >
            <a
              class="c7"
              href="/"
              target="_self"
            >
              <img
                alt="Rodrigo Silva"
                src="https://res.cloudinary.com/dccldzodg/image/upload/v1700489959/1325363_6dfc0b7add.png"
              />
            </a>
          </h1>
        </div>
        <a
          class="c8"
          href="TESTE"
          target="_self"
        >
          #TESTE
        </a>
        <a
          class="c8"
          href="TESTE2"
          target="_self"
        >
          #TESTE2
        </a>
      </nav>
    </div>
    <header
      class=""
    >
      <header
        class="c9"
      >
        <h1
          class="c5 c6"
        >
          <a
            class="c7"
            href="/"
            target="_self"
          >
            <img
              alt="Rodrigo Silva - Blog do Rodrigo"
              src="https://res.cloudinary.com/dccldzodg/image/upload/v1700489959/1325363_6dfc0b7add.png"
            />
          </a>
        </h1>
        <div
          class="c10"
        >
          <h2
            class="c5 c11"
          >
            Rodrigo Silva
          </h2>
          <p>
            Blog do Rodrigo
          </p>
        </div>
      </header>
    </header>
    <div
      class="c12"
    >
      Ol√° mundo
    </div>
    <footer
      class="c13"
    >
      <footer
        class="c14"
      >
        <div
          class="c15 c16"
        >
          texto do footer
        </div>
      </footer>
    </footer>
    <a
      aria-label="Go to top"
      class="c17"
      href="#"
      title="Go to top"
    >
      <svg
        aria-hidden="true"
        class="c1"
        fill="currentColor"
        focusable="false"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M7.41 15.41 12 10.83l4.59 4.58L18 14l-6-6-6 6 1.41 1.41z"
        />
      </svg>
    </a>
  </div>
</div>
`;
